TwitterKit/TwitterKit/TWTRTwitter.m

```diff
@@ -389,7 +389,9 @@ - (void)performWebBasedLogin:(UIViewController *)viewController completion:(TWTR
         [viewController presentViewController:navigationController animated:YES completion:nil];
     }
         completion:^(TWTRSession *session, NSError *error) {
-            [viewController dismissViewControllerAnimated:YES completion:nil];
+            // Dismiss from `presentedViewController` so that `viewController` itself doesn't get dismissed
+            // when it is also a `presentedViewController` of another presenting (parent) view controller.
+            [viewController.presentedViewController dismissViewControllerAnimated:YES completion:nil];
```

Comment:
After a bit more exploring of modal-on-modal behavior, I noticed that `[viewController.presentedViewController dismissViewControllerAnimated:YES completion:nil];` was actually unnecessary since `SFSafariViewController` will dismiss by itself on "dismiss" button tap.

After the completion of its dismissal, `viewController.presentedViewController` will mostly become `nil`.

(Disclaimer: I haven't digged other scenario of `performWebBasedLogin` though)